DEVELOPER_GUIDE.md

```diff
@@ -20,3 +24,12 @@ To run the integration tests locally, run:
 The first argument, `opensearch` tells the server type to run integration test against. Valid values are `opensearch` and `opendistro`. The second argument tells whether to run server with security plugin enabled or not.
 
 Note that integration tests require docker to be installed and running, and downloads quite a bit of data from over the internet and hence take few minutes to complete.
+
+## Build the Documentation with Sphinx
+This are the steps to build the documentation with [Sphinx](https://www.sphinx-doc.org/):
+
+1. change into the `opensearch-py` directory where `setup.py` is located
+2. install opensearch-py - we recommend [editable mode](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-e): `pip install -e .[docs]`
```

Comment:
We may also want to add a note that zsh users would need to do `pip install -e  .\[docs]` otherwise they will receive an error.

Reply:
I added doc for zsh.